### PR TITLE
Adding other KAN layers

### PR DESCRIPTION
**Pull request by armbrusl**
I have implemented the following two layers 

Julia implementation of [FourierKAN](https://github.com/GistNoesis/FourierKAN)
Julia implementation of [ChebyKAN](https://github.com/SynodicMonth/ChebyKAN)

I also added another test for all the layers. Not sure how most of GitHub works so let me know if there is anything else I should do.

### CODE REVIEW

File: .gitignore
Comment (vpuri3) [inline .gitignore:2]: This should be in your personal global `.gitignore, not here.

File: .vscode/settings.json
Comment (vpuri3) [inline .vscode/settings.json:3]: This file does not belong here. You should likely ignore `.vscode/` in the global `.gitignore` on your machine.

File: Project.toml
Comment (vpuri3) [inline Project.toml:15]: Why is Test a dependency here?

**Comment by vpuri3** [comment]
Thanks @armbrusl, the code looks ok. Can you 
- take a look at the comments
- update the README code to include the new layers

File: Project.toml
Comment (armbrusl) [reply Project.toml:17]: Due to the batched_mul function, the ChebyKAN layer will not work on the GPU without this. Is it common practice to leave this up to the user to download ?

Comment (vpuri3) [reply Project.toml:17]: Yes. You don't want to install cuTENSOR for users who aren't running on the GPU. `TensorOperations` should throw a warning if a user is calling `@tensor` with CuArrays without loading `cuTENSOR`. Can you check if TensorOperations does that? If not, you can include a warning in `batched_mul`.

**Comment by vpuri3** [comment]
just pushed a commit to master with all the Lux 0.x --> 1.x fixes. @armbrusl I also rebased this branch to master. I removed all the unnecessary packages in `test/Project.toml` (e.g., LLVM, Debugger, etc). They should not be added to the pull request. You should do a `git fetch && git pull` on your machine before proceeding. You should also update your Julia environment.

Can you also move the test you added in `test/runtests.jl` to a separate file.

File: src/cdense.jl
Comment (vpuri3) [inline src/cdense.jl:64]: same here

File: Project.toml
Comment (armbrusl) [reply Project.toml:16]: I does throw a warning

Comment (vpuri3) [reply Project.toml:16]: Great, then we don't need to do anything

**Comment by armbrusl** [comment]
I think I have addressed your suggestions. Thanks for walking me through it! Let me know if there is anything else I need to do. The only thing I haven't done is edit the README file, is there anything specific I should add ?

File: .vscode/settings.json
Comment (vpuri3) [reply .vscode/settings.json]: do a git rm on this file

File: .gitignore
Comment (vpuri3) [reply .gitignore]: revert this change.

File: Project.toml
Comment (vpuri3) [reply Project.toml]: remove cuTENSOR from Project.toml and move it to test/Project.toml

**Comment by vpuri3** [comment]
see above comments

File: src/utils.jl
Comment (vpuri3) [inline src/utils.jl:70]: Can you not use NNlib.batched_mul or NNlib.batched_vec for this?

**Review by vpuri3 — CHANGES_REQUESTED** [review]

**Comment by vpuri3** [comment]
Can you add the new layers to the code in the README and report timings?

https://github.com/vpuri3/KolmogorovArnold.jl/blob/master/README.md?plain=1#L14-L88

If not, I can do it after this is merged.

**Comment by vpuri3** [comment]
See workflow error: https://github.com/vpuri3/KolmogorovArnold.jl/actions/runs/11579597184/job/32236568354?pr=3#step:6:545

File: src/utils.jl
Comment (vpuri3) [reply src/utils.jl:70]: https://fluxml.ai/NNlib.jl/dev/reference/#NNlib.batched_mul

Comment (armbrusl) [reply src/utils.jl:70]: I am not sure. I have been trying but I have not able to get it right

**Comment by vpuri3** [comment]
@armbrusl can you investigate this error? I am not seeing it in master with 1.10 or 1.11. I also updated master to only test on 1.10 and 1.11. If you rebase this branch, then that change will be incorporated.

**Comment by armbrusl** [comment]
@vpuri3 I am not able to replicate the errors, when I run the required command 

import Pkg; Pkg.test(;coverage=true, julia_args=["--check-bounds=yes", "--compiled-modules=yes"], force_latest_compatible_version=false, allow_reresolve=true)

everything works fine. I also rebased the branch.

**Comment by vpuri3** [comment]
@armbrusl i tested the CI on Master in #4 and it is fine. There's some change introduced in this PR that is causing the failures.

I also don't see your rebase commits. Make sure you git checkout master, git fetch and git pull the latest version. then checkout this branch again and git merge master. and then do a git push

**Comment by armbrusl** [comment]
I think I have done the steps you told me, how will I know if I have done it right

**Comment by vpuri3** [comment]
I'll check out this branch and investigate this evening

**Comment by vpuri3** [comment]
@armbrusl you haven't rebased correctly. Look at [this branch](https://github.com/armbrusl/KolmogorovArnold.jl/tree/master).

image

**Pull request merged**
